src/apps/dashboard/modal/ready-for-loan-modal/ready-to-loan-modal-content.tsx

```diff
@@ -0,0 +1,237 @@
+import {
+  getColors,
+  getReadyForPickup
+} from "../../../../core/utils/helpers/general";
+import {
+  useDeleteV1UserReservationsIdentifier,
+  useGetV1UserReservations
+} from "../../../../core/publizon/publizon";
+import { useText } from "../../../../core/utils/text";
+import {
+  Reservation,
+  ReservationListResult
+} from "../../../../core/publizon/model";
+import QueuedReservationsList from "./ready-to-loan-list";
+import CheckBox from "../../../../components/checkbox/Checkbox";
+import StatusCircleIcon from "../../../loan-list/materials/utils/status-circle-icon";
+import ArrowWhite from "../../../../components/atoms/icons/arrow/arrow-white";
+import { useModalButtonHandler } from "../../../../core/utils/modal";
+
+interface ReadyToLoanModalContentProps {
+  modalId: string;
+}
+const ReadyToLoanModalContent: FC<ReadyToLoanModalContentProps> = ({
+  modalId
+}) => {
+  const t = useText();
+  const { close } = useModalButtonHandler();
+  const colors = getColors();
+  const today = dayjs();
+  const [physicalReservationsReadyToLoan, setPhysicalReservationsReadyToLoan] =
+    useState<ReservationDetailsV2[]>();
+  const [digitalReservationsReadyToLoan, setDigitalReservationsReadyToLoan] =
+    useState<Reservation[]>();
+  const { data: physicalReservations } = useGetReservationsV2();
+  const { data: digitalReservations } =
+    useGetV1UserReservations<ReservationListResult>();
+  const [allSelectableReservations, setAllSelectableReservations] = useState<
+    {
+      [key: string]: string;
+    }[]
+  >();
+  const [selectedReservations, setSelectedReservations] = useState<
+    {
+      [key: string]: string;
+    }[]
+  >([]);
+  const { mutate: deletePhysicalReservation } = useDeleteReservations();
+  const { mutate: deleteDigitalReservation } =
+    useDeleteV1UserReservationsIdentifier();
+  useEffect(() => {
+    if (physicalReservations && !physicalReservationsReadyToLoan) {
+      const readyToLoan = getReadyForPickup(physicalReservations);
+      if (readyToLoan) {
+        setPhysicalReservationsReadyToLoan(readyToLoan);
+      }
+    }
+  }, [physicalReservations, physicalReservationsReadyToLoan, today]);
+
+  useEffect(() => {
+    if (digitalReservations && !digitalReservationsReadyToLoan) {
+      const { reservations } = digitalReservations;
+      if (reservations) {
+        const readyToLoan = reservations.filter((reservation) => {
+          const expiryDate = dayjs(reservation.expireDateUtc);
+          return expiryDate > today;
+        });
+        if (readyToLoan) {
+          setDigitalReservationsReadyToLoan(readyToLoan);
+        }
+      }
+    }
+  }, [digitalReservations, digitalReservationsReadyToLoan, today]);
+
+  useEffect(() => {
+    if (
+      physicalReservationsReadyToLoan &&
+      digitalReservationsReadyToLoan &&
+      !allSelectableReservations
+    ) {
+      const fausts = physicalReservationsReadyToLoan.reduce((acc, pr) => {
```

Comment:
You can also write:
```
      const fausts = physicalReservationsReadyToLoan.reduce<{[key: string]: string}>((acc, pr) => {
```
Which goes for both reduces.
Then you might be able to skip the type assertion at line 98